apollo-router-core/src/query_cache.rs

```diff
@@ -18,7 +21,8 @@ struct QueryCacheResolver {
 impl CacheResolver<String, Option<Arc<Query>>> for QueryCacheResolver {
     async fn retrieve(&self, key: String) -> Result<Option<Arc<Query>>, CacheResolverError> {
         let schema = self.schema.clone();
-        let query_parsing_future = tokio::task::spawn_blocking(move || Query::parse(key, &schema));
+        let query_parsing_future = tokio::task::spawn_blocking(move || Query::parse(key, &schema))
+            .instrument(info_span!("parse_query", "otel.kind" = %SpanKind::Internal));
```

Comment:
I don't know if this is a problem or not, but won't we now have two spans called "parse_query"?

Reply:
I don't know either. Maybe discuss tomorrow?